libyara/modules/pe.c

```diff
@@ -1849,6 +1853,52 @@ define_function(rich_toolid_version)
       rich_internal(module(), integer_argument(2), integer_argument(1)));
 }
 
+
+define_function(calculate_checksum)
+{
+  YR_OBJECT* module = module();
+
+  PE* pe = (PE*) module->data;
+  if (pe == NULL)
+    return_integer(UNDEFINED);
+
+  uint64_t csum = 0;
```

Comment:
In C99 variables can be declared anywhere within a block, but in C89 they must be at the beginning of the block. This has been a source of problems with Visual Studio 2010.